pkg/registry/apis/alerting/notifications/receiver/storage.go

```diff
@@ -35,7 +37,8 @@ func NewStorage(
 	scheme *runtime.Scheme,
 	desiredMode grafanarest.DualWriterMode,
 	optsGetter generic.RESTOptionsGetter,
-	reg prometheus.Registerer) (rest.Storage, error) {
+	reg prometheus.Registerer,
+	serverLockService *serverlocksvc.ServerLockService) (rest.Storage, error) {
```

Comment:
The growth of dual write dependencies seems fishy -- perhaps we can restructure this so it is a single parameter that can hold/hide the range of dependencies?

Pehraps `desiredMode`, `reg`, and `serverLockService` can all be replaced with a function supplier?  getDualWriter(legacy) -- so the details of constructing a dual writer do not need to be copied to the growing list of resources that will need one?

Reply:
> perhaps we can restructure this so it is a single parameter that can hold/hide the range of dependencies?

That's a good point!

I have refactored the dependencies into a single `DualWriterOptions` struct, so the NewDualWriter function will be like this

```
func NewDualWriter(
	legacy LegacyStorage,
	storage Storage,
	options DualWriterOptions,
) DualWriter
```

Let me know if you want me to change this.

Reply:
the part I am more concerned with is that all the parameters must be injected into each apiserver and then constructed manually.  Seems better if we can just provide a function that creates it for you and then each service does not need to evolve as this keeps changing